internal/source/source.go

```diff
@@ -0,0 +1,31 @@
+package source
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Source struct {
+	src    string
+	reader *strings.Reader
+}
+
+func New(src string) *Source {
+	return &Source{
+		src:    src,
+		reader: strings.NewReader(src),
+	}
+}
+
+func (s Source) Reader() *strings.Reader {
+	return strings.NewReader(s.src)
+}
+
+func (s *Source) Snippet(pos int) string {
```

Comment:
Can you please add documentation with examples to this method?

Reply:
Also some unit tests

Reply:
It's used to decorate an error to a pointer to position at the expression, this is one of the unit test form lexer:
```
invalid number format (f is not a digit), pos=6
|  foo 1foo 
| ......^
```

The `Snippet` decorate "invalid number format" error with the expression and a pointer to the position that is invalid.